clang/lib/Sema/HLSLExternalSemaSource.cpp

```diff
@@ -67,6 +67,7 @@ struct BuiltinTypeDeclBuilder {
     Record = CXXRecordDecl::Create(AST, TagDecl::TagKind::Class, HLSLNamespace,
                                    SourceLocation(), SourceLocation(), &II,
                                    PrevDecl, true);
+    Record->startDefinition();
```

Comment:
We should not be starting the definition here - this constructor just creates the declaration.